src/classes/Entity.ts

```diff
@@ -758,7 +758,7 @@ class Entity<
 
     // Check for required fields
     Object.keys(required).forEach(field =>
-      required[field] && !data[field]
+      required[field] && (data[field] === undefined || data[field] === null)
```

Comment:
The only exception would be `NaN`.

From [MDN](https://developer.mozilla.org/en-US/docs/Glossary/NaN):

> NaNs can represent infinity, result of division by zero, missing value, or the square root of a negative (which is imaginary, whereas a floating-point number is real).

Should the library allow `NaN` to be persisted? Or should that be treated as a value?

Reply:
`NaN` should not be persisted, so perhaps I'll add a check. Going to merge this because people are asking for it.